pkgs/development/python-modules/undetected-chromedriver/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, selenium
+, requests
+, websockets
+}:
+
+buildPythonPackage rec {
+  pname = "undetected-chromedriver";
+  version = "3.5.4";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "ultrafunkamsterdam";
+    repo = pname;
+    rev = "783b8393157b578e19e85b04d300fe06efeef653";
+    hash = "sha256-vQ66TAImX0GZCSIaphEfE9O/wMNflGuGB54+29FiUJE=";
+  };
+
+  disabled = pythonOlder "3.7";
+
+  propagatedBuildInputs = [
+    selenium
+    requests
+    websockets
+  ];
+
+  pythonImportsCheck = [
+    "undetected_chromedriver"
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/UltrafunkAmsterdam/undetected-chromedriver";
+    changelog = "https://github.com/ultrafunkamsterdam/undetected-chromedriver/blob/master/README.md";
+    license = licenses.gpl3Only;
+    description = ''
+      Selenium.webdriver.Chrome replacement with compatiblity for Brave, and other Chromium based browsers.
+    '';
+    maintainers = with maintainers; [ liammurphy14 ];
+  };
```

Comment:
Not sure whether this should specify `sourceProvenance = [ sourceTypes.binaryNativeCode ]` if the software downloads a binary driver (https://github.com/ultrafunkamsterdam/undetected-chromedriver/blob/master/README.md?plain=1#L7).

@amjoseph-nixpkgs @risicle ?

Reply:
This... is a case we never considered, because normally nix packages try to make themselves self-contained. A package that downloads binaries at runtime and runs them is a bit of an anathema to nix packaging, though I realize this package has Reasons for doing it.

For now I would personally mark this as `binaryNativeCode` with a comment explaining the weirdness. That probably most closely captures the intent of most `NIXPKGS_ALLOW_NONSOURCE` users.

Reply:
Sounds good, I will do that. Thanks for the reviews!